src/core/plugin/luapi_application.h

```diff
@@ -1402,6 +1403,43 @@ static int applib_changeCurrentPageBackground(lua_State* L) {
     return 0;
 }
 
+/**
+ * Query all colors of the current palette.
+ *
+ * Example1: palette = app.getColorPalette()
+ * possible return value:
+ * {
+ *     [1] = {color = 0xffffff, name = "white" },
+ *     [2] = {color = 0x000000, name = "black" },
+ *     [3] = {color = 0xdc8a78, name = "Rosewater" },
+ * }
+ *
+ */
+static int applib_getColorPalette(lua_State* L) {
+    // discard any extra arguments passed in
+    lua_settop(L, 0);
+
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Settings* settings = plugin->getControl()->getSettings();
+    const Palette& palette = settings->getColorPalette();
+
+
+    lua_newtable(L);  // return table
+    for (size_t i{0}; i < palette.size(); i++) {
+        lua_pushinteger(L, static_cast<lua_Integer>(i) + 1);  // index in return table
+        lua_newtable(L);                                      // item table
+
+        const auto& col = palette.getColorAt(i);
+        lua_pushstring(L, col.getName().c_str());
+        lua_setfield(L, -2, "name");
+        lua_pushinteger(L, uint32_t(col.getColor()));
```

Comment:
We don't want RGBA, but only RGB
```suggestion
        lua_pushinteger(L, int(uint32_t(col.getColor()) & 0xffffffU));
```

Reply:
Oh didn't think about that one, thanks for pointing out.

Wouldn't be `as_signed` be more consistent?

Reply:
Agree, `as_signed` is better. We have the cast to `int` for colors on multiple places in the code file though, so that would have to be changed accordingly.